### PR TITLE
fix(MCDU): PERF APPR no longer demands QNH and TEMP when >180NM from destination

### LINKED ISSUE #8126: FMS appr: Temperature and QNH fields mandatory when distance to dest greater than 180NM
### Aircraft Version

Development

### Build info

```json
{
    "built": "2023-07-20T17:51:47+00:00",
    "ref": "refs/heads/master",
    "sha": "3f1edb89e14625a0b8186082d722a58f3542338a",
    "actor": "Gurgel100",
    "event_name": "manual",
    "pretty_release_name": "master:3f1edb89",
    "version": "v0.11.0-dev.3f1edb8"
}
```

### Describe the bug

In the FMS approach page, QNH and temperature fields appear in mandatory amber boxes instead of brackets when the airplane is more than 180NM from the programmed destination.
PROG
APPR
F-PLN

### Expected behavior

QNH and temperature fields should only be shown in mandatory amber boxes, if distance to destination is less than 180NM.

### Steps to reproduce

1) Observe the performance approach page when more than 180NM away from the destination airport.

### References (optional)

image

### Additional info (optional)

_No response_

### Discord Username (optional)

Bruno_pt99#5802

### PR DESCRIPTION
**Pull request by robertxing2004**
Fixes #8126 

## Summary of Changes

Corrected closeToDest logic so that the MCDU PERF APPR page no longer demands QNH and TEMP for arrival airport when >180NM from destination.

## References

image

## Additional context

Discord username (if different from GitHub): robeet

## Testing instructions

- Spawn aircraft
- Observe PERF APPR page in the following states, performing flight as necessary:
- No destination (INIT A FROM/TO empty)
- over 180NM from destination
- within 180NM from destination

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

**Comment by BlueberryKing** [comment]
Thanks for the quick fix! @BravoMike99 can correct me, but I think the temperature field should always be shown as a cyan box, even if closer than 180NM. Maybe this can be changed in the PR as well?

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_PerformancePage.js [hunk 1]
```diff
@@ -706,7 +706,7 @@ class CDUPerformancePage {
             }
         };
 
-        const closeToDest = mcdu.flightPlanManager.getDestination() && mcdu.flightPlanManager.getDestination().liveDistanceTo <= 180;
+        const closeToDest = mcdu.flightPlanManager.getDistanceToDestination() != -1 && mcdu.flightPlanManager.getDistanceToDestination() <= 180;
```
Comment (BlueberryKing) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_PerformancePage.js:709]:
```suggestion
        const closeToDest = mcdu.flightPlanManager.getDistanceToDestination() !== -1 && mcdu.flightPlanManager.getDistanceToDestination() <= 180;
```

**Review by BlueberryKing — APPROVED** [review]
LGTM

**Review by tracernz — APPROVED** [review]

**Comment by tracernz** [comment]
Technically it's also demanding the wind, even though it doesn't turn amber. Chanelog is fine as-is though.

File: fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_PerformancePage.js [hunk 2]
```diff
@@ -706,7 +706,7 @@ class CDUPerformancePage {
             }
         };
 
-        const closeToDest = mcdu.flightPlanManager.getDestination() && mcdu.flightPlanManager.getDestination().liveDistanceTo <= 180;
+        const closeToDest = mcdu.flightPlanManager.getDistanceToDestination() !== -1 && mcdu.flightPlanManager.getDistanceToDestination() <= 180;
```
Comment (tracernz) [inline fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_PerformancePage.js:709]:
```suggestion
        const distanceToDestination = mcdu.flightPlanManager.getDistanceToDestination();
        const closeToDest = distanceToDestination >= 0 && distanceToDestination <= 180;
```
The call is relatively expensive, so maybe just call once.

**Comment by auroraisluna** [comment]
QA Report

Discord: alepouna
Object of testing: #8159
Tier of Testing : 1
Date : 19/08/2023

Testing Process:
Observe PERF APPR page in the following states, performing flight as necessary:
- [PASS] No destination (INIT A FROM/TO empty)
- [PASS] over 180NM from destination
- [PASS] within 180NM from destination

Testing Results:
Passed

**Pull request merged**